### DISCUSSION TITLE
Problems using Countdown Timer.

### DISCUSSION THREAD

**Discussion by RupertRiddle** [jspsych/jsPsych#2690] [Q&A] [answered]
Hi guys,

I came across a countdown timer previously implemented by someone else on the forum, linked here: https://github.com/jspsych/jsPsych/discussions/1690

I am trying to integrate it into my code, but I get a typeError which says:
 TypeError: trial.stimulus is not a function

Any ideas how to resolve this issue? 

Best Wishes,
Rupert

**Comment by jodeleeuw** [comment, marked as answer]
Can you share a little more about the code you are using? Are you working with jsPsych v6 or v7?

**Reply by RupertRiddle** [reply]
Hi Joe thanks for getting back to me!

I'm using V7. I changed a few small things from the post, the exact code is:


`
var delay = {
  type: jsPsychCanvasButtonResponse,
  stimulus: '<p>The next part of the experiment will start in <span id="clock">1:00</span>.',
  choices: ['Continue'],
  on_load: function(){
    var wait_time = 1 * 60 * 1000; // in milliseconds
    var start_time = performance.now();
    document.querySelector('button').disabled = true;
    var interval = setInterval(function(){
      var time_left = wait_time - (performance.now() - start_time);
      var minutes = Math.floor(time_left / 1000 / 60);
      var seconds = Math.floor((time_left - minutes*1000*60)/1000);
      var seconds_str = seconds.toString().padStart(2,'0');
      document.querySelector('#clock').innerHTML = minutes + ':' + seconds_str
      if(time_left <= 0){
        document.querySelector('#clock').innerHTML = "0:00";
        document.querySelector('button').disabled = false;
        clearInterval(interval);
      }
    }, 250)
  }
}








`

**Reply by jodeleeuw** [reply]
The reason you are running into that error message is that you switched from the html-button-response plugin to the canvas-button-response plugin. The canvas plugin expects the `stimulus` property [to be a function that will draw something on a canvas](https://www.jspsych.org/7.2/plugins/canvas-button-response/). Try changing the `type` to `jsPsychHtmlButtonResponse`.

**Reply by RupertRiddle** [reply]
Thanks that solved the problem - My mistake :)
